### ISSUE TITLE
Samples could be corrupted! Reduce frequency

### ISSUE THREAD

**Issue by andrasfuchs** [7m4mon/gr-ad2#4]
I opened [AnalogIn_Record_FM_Demon.drc](https://github.com/7m4mon/gr-ad2/blob/master/AnalogIn_Record_FM_Demod.grc) on Ubuntu, and executed the flow graph in GRC 3.7.13.4.

If I start increasing the samp_rate variable in the flow chart, I start getting the "Samples could be corrupted! Reduce frequency" error from 512k. 
In WaveForms' Scope I can see the Analog Discovery 2 can sample at up to 100 MHz.

It would be practical to make this setting work with a value of at least 4 MHz so that we could receive the AM radio stations in the area.

The log from the run with 512k samp_rate setting.
> <<< Welcome to GNU Radio Companion 3.7.13.4 >>>
> 
> Block paths:
> 	/usr/share/gnuradio/grc/blocks
> 	/usr/local/share/gnuradio/grc/blocks
> 
> Loading: "/home/ubuntu/Work/gr-ad2/AnalogIn_Record_FM_Demod.grc"
> Done
> 
> Generating: '/home/ubuntu/Work/gr-ad2/top_block.py'
> Warning: This flow graph contains a throttle block and another rate limiting block, e.g. a hardware source or sink. This is usually undesired. Consider removing the throttle block.
> 
> Executing: /usr/bin/python2 -u /home/ubuntu/Work/gr-ad2/top_block.py
> 
> Warning: failed to XInitThreads()
> gr::log :INFO: audio source - Audio sink arch: alsa
> DWF Version: 3.12.2
> BufferSize 16384
> begin acquisition
> cCorrupted3394
> cCorrupted3636
> cCorrupted3961
> 
> (top_block.py:8268): Gtk-WARNING **: 21:27:56.361: Negative content width -13 (allocation 1, extents 7x7) while allocating gadget (node button, owner GtkButton)
> 
> (top_block.py:8268): Gtk-WARNING **: 21:27:56.361: Negative content height -5 (allocation 1, extents 3x3) while allocating gadget (node button, owner GtkButton)
> aUcCorrupted4332
> aUcCorrupted3763
> aUcCorrupted4070
> cCorrupted4463
> cCorrupted1454
> cCorrupted705
> aUcCorrupted3273
> aUcCorrupted3282
> cCorrupted3536
> cCorrupted3204
> cCorrupted2758
> cCorrupted1703
...
> aUcCorrupted15244
> aUaUcCorrupted14697
> aUcCorrupted15768
> aUcCorrupted46
> aUaUaUcCorrupted15202
> finished
> Samples could be corrupted! Reduce frequency
> 
> Done

**Comment by 7m4mon** [comment]
The AnalogDiscovery2 has a capability of 100MHz sampling.
But, due to limitation of PC performance or transmission rate of USB 2.0, real-time processing rates are reduced.
Even if you only record using WAVEFORM application, the transmission rate is up to 1MHz.

**Comment by andrasfuchs** [comment]
I see, so does this mean that we have no chance to get AM and FM radio stations with AD2 + GNU Radio?

In my country AM stations start at 540 kHz and FM stations end at 104.9 MHz.

**Comment by 7m4mon** [comment]
AM radio listening may be possible with under sampling.  
I assembled the tuning circuit (Bar coils and valiable capacitor), and connected to antenna (about 10m wire in outdoor).  
![tuning circuit for AM radio](https://user-images.githubusercontent.com/11815644/74607570-4b73c080-511d-11ea-83f9-7539eebfe85f.jpg)  
I was able to hear a AM broadcasting signal (810kHz, 9th nyquist zone with 192ksps) with GRC. (but it was very weak)  
![AnalogIn_Record_AM_Demod grc](https://user-images.githubusercontent.com/11815644/74607586-6a725280-511d-11ea-9893-0ee7b67a6cc6.png)  
If the target frequency is around the Nyquist frequency(1/2 Fs), it will not be received properly.  

FM radio sampling with AD2, it seems to difficult. Because AD2's input bandwidh is < 30MHz.   
I think it needs frequency conversion like heterodyne receiver.
